scene/gui/tree.cpp

```diff
@@ -701,12 +701,35 @@ void TreeItem::set_visible(bool p_visible) {
 		tree->queue_redraw();
 		_changed_notify();
 	}
+
+	_handle_visibility_changed(p_visible);
 }
 
 bool TreeItem::is_visible() {
 	return visible;
 }
 
+bool TreeItem::is_visible_in_tree() const {
+	return visible && parent_visible_in_tree;
+}
+
+void TreeItem::_handle_visibility_changed(bool p_visible) {
+	TreeItem *child = get_first_child();
+	while (child) {
+		child->_propagate_visibility_changed(p_visible);
+		child = child->get_next();
+	}
+}
+
+void TreeItem::_propagate_visibility_changed(bool p_parent_visible_in_tree) {
+	if (p_parent_visible_in_tree == parent_visible_in_tree || !visible) {
```

Comment:
This is wrong. `parent_visible_in_tree` should be set unconditionally, like in CanvasItem.
Unless there is a reason for this code?

Reply:
Thanks for the review. I removed the condition.

The `!visible` part was wrong. For the first part, I think that the state of any sub tree won't change if its root node already has the right state of  `parent_visible_in_tree`. Unless a future code adds node manually, and doesn't take care of the setting `parent_visible_in_tree` correctly.
Anyway, that's not necessary to validate the code. Furthermore, if the propagation function becomes more complex, the condition could not be correct anymore.
I'll be careful next time for this kind of thing.